packages/core/src/utils.ts

```diff
@@ -196,9 +196,30 @@ export function isNotString<T>(value: string | T): value is T {
   return !isString(value);
 }
 
-export function assignExisting<T>(target: Record<string, T>, obj: Record<string, T>) {
+export const assignRules = <T extends string | { severity?: string }>(
```

Comment:
I see it's being applied to decorators and preprocessors as well. Could you come up with a more generic/descriptive name?